### PR TITLE
Fix close order for FSContext resources

### PR DESCRIPTION
**Pull request by ggezer**
Netty event loop manages propagating of stream control messages. Alluxio's authentication streams are bound to life-time of logical channels and it's unexpected for a stream to loose its loop while still alive.

### CODE REVIEW

**Comment by ggezer** [comment]
Jenkins, test this please

File: core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java [hunk 1]
```diff
@@ -259,14 +259,14 @@ private synchronized void closeContext() throws IOException {
...
       mClosed.set(true);
-      mWorkerGroup.shutdownGracefully(1L, 10L, TimeUnit.SECONDS);
       mFileSystemMasterClientPool.close();
       mFileSystemMasterClientPool = null;
       mBlockMasterClientPool.close();
       mBlockMasterClientPool = null;
       for (BlockWorkerClientPool pool : mBlockWorkerClientPool.values()) {
         pool.close();
       }
+      mWorkerGroup.shutdownGracefully(1L, 10L, TimeUnit.SECONDS);
```
Comment (gpang) [inline core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java]: Can you add a comment in the code of what the order should be? (probably a simple comment similar to the PR description?)

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please

**Pull request merged**